Checkpoint 2

anarianna657-bit opened the pull request.

brainark1 committed:
Add checkpoint 1 test

anarianna657-bit committed:
Update build.gradle

brainark1 committed:
test git metadata removal

anarianna657-bit committed:
Merge branch 'main' of https://github.com/CPS353-Suny-New-Paltz/project-starter-code-anarianna657-bit

anarianna657-bit committed:
test all is well

anarianna657-bit committed:
wrong code

anarianna657-bit committed:
right code

anarianna657-bit committed:
Merge pull request #2 from CPS353-Suny-New-Paltz/newcode2

test all is well

anarianna657-bit committed:
Update TestStatusCheckPR.java

anarianna657-bit committed:
Merge branch 'main' of https://github.com/CPS353-Suny-New-Paltz/project-starter-code-anarianna657-bit

anarianna657-bit committed:
Update TestStatusCheckPR.java

anarianna657-bit committed:
Merge pull request #5 from CPS353-Suny-New-Paltz/checkpoint-1-tests

part 6

anarianna657-bit committed:
Update README.md

anarianna657-bit committed:
UserComputeAPI

anarianna657-bit committed:
Merge branch 'main' of https://github.com/CPS353-Suny-New-Paltz/project-starter-code-anarianna657-bit

anarianna657-bit committed:
StorageComputeAPI

anarianna657-bit committed:
EngineAPI

anarianna657-bit committed:
metadata

anarianna657-bit committed:
Update workbench.xmi

anarianna657-bit committed:
more metadata

anarianna657-bit committed:
go away

anarianna657-bit committed:
idk

anarianna657-bit committed:
edits

anarianna657-bit committed:
UserAPI

UserAPI and its prototype: set input and output.

anarianna657-bit committed:
update check2

anarianna657-bit committed:
StorageAPI

Read input and write output

anarianna657-bit committed:
EngineAPI

works with ConceptualAPI to print result

anarianna657-bit committed:
Main

Test program

anarianna657-bit committed:
Update checkstyle_checks.xml (#8)

bnavs706 reviewed (approved):
Everything looks good runs as expected!

anarianna657-bit committed:
all my apis recommitted (#9)

so i can merge to main

anarianna657-bit committed:
part 5 (#4)

* incorrect code

* right code

* incorrect again

* right code again

* delete restrictions

anarianna657-bit committed:
Delete WorldQueries/src directory (#14)

anarianna657-bit committed:
Remove TypeName module from checkstyle configuration (#13)

anarianna657-bit committed:
updated folders (#10)

* updated folders

* annotated prototypes

* delete stuff

* put all in project.annotations

* updates

* update 2

anarianna657-bit committed:
Fix (#16)

* hjb

* Update TestStatusCheckPR.java

* update

* hb

* Update TestStatusCheckPR.java

* Update TestStatusCheckPR.java

* Update TestStatusCheckPR.java

* Update TestStatusCheckPR.java

anarianna657-bit committed:
Update TestStatusCheckPR.java (#24)

anarianna657-bit committed:
Update checkstyle_checks.xml (#26)

anarianna657-bit committed:
Update TestStatusCheckPR.java (#29)

anarianna657-bit committed:
Update TestStatusCheckPR.java (#31)

anarianna657-bit committed:
Update TestStatusCheckPR.java (#34)

anarianna657-bit committed:
Delete test/TestStatusCheckPR.java (#35)

anarianna657-bit committed:
Update TestStatusCheckPR.java (#36)

anarianna657-bit committed:
Update TestStatusCheckPR.java (#37)

anarianna657-bit committed:
skeletonSpooky1 (#39)

* skeletonSpooky1

* added to main, will edit later

* debugged

anarianna657-bit committed:
InMemory (#42)

Input, Output, StorageAPI (Step 4)

anarianna657-bit committed:
In memory (#43)

* InMemory

Input, Output, StorageAPI (Step 4)

* Update InMemoryStorageAPI.java

anarianna657-bit committed:
In memory (#45)

* InMemory

Input, Output, StorageAPI (Step 4)

* Update InMemoryStorageAPI.java

* Create ComputeEngineIntegrationTest.java

step 5 check3

anarianna657-bit committed:
update In memory (#46)

* InMemory

Input, Output, StorageAPI (Step 4)

* Update InMemoryStorageAPI.java

* Create ComputeEngineIntegrationTest.java

step 5 check3

* Update ComputeEngineIntegrationTest.java

* Update ComputeEngineIntegrationTest.java

anarianna657-bit committed:
Update build.gradle (#48)

anarianna657-bit committed:
Check3 (#49)

* Update build.gradle

* Create Checkpoint3TestSuite.java

* Update Checkpoint3TestSuite.java

* Update build.gradle

anarianna657-bit committed:
Create ComputationComponent.java (#50)

* Create ComputationComponent.java

Step one of checkpoint 4 works with conceptual and follow declared computation

* Update package name

moving checkpoint4 into projects to attempt to pass the checkstyle

* Update ComputationComponent.java

fixing curly brackets to pass checkstyle

anarianna657-bit committed:
4.1 (#51)

* Create ComputationComponent.java

Step one of checkpoint 4 works with conceptual and follow declared computation

* Update package name

moving checkpoint4 into projects to attempt to pass the checkstyle

* Update ComputationComponent.java

fixing curly brackets to pass checkstyle

* Update ComputationComponent.java

remove abstract

anarianna657-bit committed:
Create Coordination and DataStorage (#52)

Steps 2 and 3 committed together because they work together

anarianna657-bit committed:
tests (#41)

* tests

not expected to pass yet because of skeleton implementations

* Remove filter on build.gradle

* Remove filter on build.gradle

* remove .* on mockito import

to pass checkstyle

* remove abstract

making implementations concrete to pass gradle

* calculatePrimes

adding to engine api and its implementation

* Update build.gradle

adding filter back because I removed it too soon

* Update UserAPIImpl.java

* Update TestEngineAPI.java

update import

* Update EngineAPIImpl.java

* Update EngineAPIImpl.java

* Update ComputationComponent.java

* Update EngineAPIImpl.java

* Update EngineAPIImpl.java

* Update UserAPIImpl.java

* Update UserAPIImpl.java

* Update EngineAPIImpl.java

* trying to fix gradle again

* Update build.gradle

* Update build.gradle

* Update TestEngineAPI.java

* Update build.gradle

* Create UserEngineIntegrationTest.java

creating the integration test

* update integration test

* update build.gradle

* Update build.gradle

* Update UserEngineIntegrationTest.java

* Update UserEngineIntegrationTest.java

anarianna657-bit committed:
steps 4 and 5 (#54)

adding an additional test to TestUserAPI to satisfy step 5. Removing filter from build.gradle to satisfy step 4

anarianna657-bit committed:
4(6/7) (#55)

* steps 4 and 5

adding an additional test to TestUserAPI to satisfy step 5. Removing filter from build.gradle to satisfy step 4

* steps 4 and 5

adding an additional test to TestUserAPI to satisfy step 5. Removing filter from build.gradle to satisfy step 4

* steps 6

* steps 6

* Update StorageAPIImpl.java

* Update ManualTestingFramework.java

anarianna657-bit committed:
Update system diagram image in README (#56)

Replaced the previous system diagram image with a new screenshot.

anarianna657-bit committed:
getting rid of all void (#57)

* getting rid of all void

Based off of grading comments, I need to get rid of any void methods so. that is what I am doing here and cleaning things up.

* fixing gradle errors

* Update InMemoryStorageAPI.java

anarianna657-bit committed:
Corrections  (#58)

* getting rid of all void

Based off of grading comments, I need to get rid of any void methods so. that is what I am doing here and cleaning things up.

* fixing gradle errors

* Update InMemoryStorageAPI.java

* checkpoint 4 corrections

coding corrections

* Update StorageAPIImpl.java

* Update StorageAPIImpl.java

* Update UserAPIImpl.java

* Update ManualTestingFramework.java

* Create manualTestOutput.txt

* output finally printed

anarianna657-bit committed:
Update image in README with new screenshot (#59)

anarianna657-bit committed:
step 1 of checkpoint 5 (#60)

* step 1 of checkpoint 5

adding validations

* Update ComputationComponent.java

* Update ComputationComponent.java

anarianna657-bit committed:
Step 2 of 5 (#61)

* Step 2 of 5

add exception handling

* Improve readability of isPrime method

Refactor isPrime method for improved readability.

* Refactor prime check loop for better readability

anarianna657-bit committed:
Step 3 of 5 (#62) validation test added

* Step 2 of 5

add exception handling

* Update ComputationValidationTest.java

anarianna657-bit committed:
Create ExceptionHandlingIntegrationTest.java (#63)

integration test for step 4

anarianna657-bit committed:
Adding Checkpoint5TestSuite.java (#64)

adding the test suite

anarianna657-bit committed:
Update image in README with new screenshot (#65)

* Update image in README with new screenshot

* Update README to remove outdated screenshot

Removed an outdated screenshot from the README.

anarianna657-bit committed:
Update image in README with new screenshot (#66)

anarianna657-bit committed:
corrections for checkpoint 5 (#67)

* corrections for checkpoint 5

fixing up StorageAPI and all things affected by it

* Update ComputeEngineIntegrationTest.java

anarianna657-bit committed:
steps a-c (#68)

* steps a-c

fixing the todos in testuser and testmultiuser

* fixing checkstyle errors

* Update TestUser.java

anarianna657-bit committed:
1/6 (#69)

* steps a-c

fixing the todos in testuser and testmultiuser

* fixing checkstyle errors

* Update TestUser.java

* steps d and e

this is the input file that completes step 1

anarianna657-bit committed:
2/6 (#70)

* part 2 of checkpoint 6

multithreaded implementation

* Update MultiThreadedCoordinationComponent.java

anarianna657-bit committed:
Update README with thread selection rationale (#71)

Added explanation for thread selection in multithreaded compute engine.

anarianna657-bit committed:
2/6  (#72)

* part 2 of checkpoint 6

multithreaded implementation

* Update MultiThreadedCoordinationComponent.java

* updated testharness files

updated to work with multithreadedcoordinationcomponent

anarianna657-bit committed:
Configure Checkstyle task reports in build.gradle (#73)

ReviewDog annotations for checkstyle

anarianna657-bit committed:
Add Checkstyle plugin to build.gradle (#75)

* Add Checkstyle plugin to build.gradle

* Add Checkstyle configuration to build.gradle

* Refactor build.gradle for checkstyle configuration

* Add source sets for main and test Java directories

anarianna657-bit committed:
Add Checkstyle workflow with Reviewdog integration (#74)

* Add Checkstyle workflow with Reviewdog integration

* Update reviewdog action version in workflow

* Refactor Checkstyle workflow for ReviewDog integration

* Update Checkstyle command to include main and test

anarianna657-bit committed:
(Final) Add ReviewDog Checkstyle integration details (#76)

Added section about ReviewDog Checkstyle integration to README.

anarianna657-bit committed:
Remove accidentally committed output file (#78)

* Remove accidentally committed output file

* Update UserAPIImpl.java

* Update UserAPIImpl.java

* Update UserAPIImpl.java

anarianna657-bit committed:
Refactor build.gradle to use plugins block (#79)

checkpoint 7 gRPC and protobuf dependencies added

anarianna657-bit committed:
Refactor build.gradle for plugin and dependency management (#80)

anarianna657-bit committed:
Refactor build.gradle to use plugins block (#81)

anarianna657-bit committed:
coordinator.proto(check 7) (#83)

* Remove accidentally committed output file

* Update UserAPIImpl.java

* Update UserAPIImpl.java

* Update UserAPIImpl.java

* Add Coordinator gRPC service definition

* Update build.gradle

* Create settings.gradle

anarianna657-bit committed:
Check7 (#84)

* Remove accidentally committed output file

* adding java

anarianna657-bit committed:
Add dependency on generateProto task for compileJava (#85)

anarianna657-bit committed:
Check7 (#86)

* Remove accidentally committed output file

* adding java

* datastore

* Update build.gradle

anarianna657-bit committed:
Check8 (#87)

* Remove accidentally committed output file

* benchmark test and slow and fast impl

anarianna657-bit committed:
Document performance improvements in README (#88)

Added section on performance improvements with caching for prime number calculations.

anarianna657-bit committed:
Check6corrections (#89)

* Remove accidentally committed output file

* adding validation

* fix checkstyle error

* Update UserEngineIntegrationTest.java
